control_panel_api/views.py

```diff
@@ -48,3 +49,14 @@ class S3BucketViewSet(viewsets.ModelViewSet):
     serializer_class = S3BucketSerializer
     filter_backends = (S3BucketFilter,)
     permission_classes = (S3BucketPermissions, )
+
+    def perform_create(self, serializer):
+        instance = serializer.save()
+        services.create_bucket(instance.name)
+        services.create_bucket_policies(instance.name)
+
+    def perform_destroy(self, instance):
+        name = instance.name
+        instance.delete()
+        services.delete_bucket_policies(name)
+        services.delete_bucket(name)
```

Comment:
@xoen here did you say NOT delete bucket? Could also add a settings flag to enable or disable